langchain/src/smith/runner_utils.ts

```diff
@@ -534,23 +534,22 @@ const applyEvaluators = async ({
     total: examples.length,
     format: "Running Evaluators: {bar} {percentage}% | {value}/{total}\n",
   });
-  const results: Record<
-    string,
-    { run_id: string; execution_time?: number; feedback: Feedback[] }
-  > = {};
-  for (let i = 0; i < runs.length; i += 1) {
-    const run = runs[i];
-    const example = examples[i];
+
+  const results = await Promise.all(runs.map(async (run, i): Promise<{
```

Comment:
Wouldn't this run all evaluators in parallel over the entire dataset, with no semaphor (max_concurrency)?

That would likely lead to flakiness. We should respect the concurrency setttings passed in to the `runOnDataset` function

Reply:
We should be able to instead wrap this inside an instance of [`AsyncCaller`](https://api.js.langchain.com/classes/langchain_core_utils_async_caller.AsyncCaller.html) which we can pass `maxConcurrency` to